perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,838 @@
+
+# Map from FLAGs.mysql_svc_db_instance_cores to RDS DB Type
+RDS_CORE_TO_DB_CLASS_MAP = {
+    '1': 'db.m3.medium',
+    '4': 'db.m3.xlarge',
+    '8': 'db.m3.2xlarge',
+    '16': 'db.r3.4xlarge',  # m3 series doesn't have 16 core.
+}
+
+RDS_DB_ENGINE = 'MySQL'
+RDS_DB_ENGINE_VERSION = '5.6.23'
+RDS_DB_STORAGE_TYPE_GP2 = 'gp2'
+# Storage IOPS capacity of the DB instance.
+# Currently this is fixed because the cloud provider GCP does not support
+# changing this setting. As soon as it supports changing the storage size, we
+# will expose a flag here to allow caller to select a storage size.
+# Default GCP storage size is 1TB PD-SSD which supports 10K Read or 15K Write
+# IOPS (12.5K mixed).
+# To support 12.5K IOPS on EBS-GP, we need 4170 GB disk.
+RDS_DB_STORAGE_GP2_SIZE = '4170'
+
+# Constants defined for Sysbench tests.
+RAND_INIT_ON = 'on'
+DISABLE = 'disable'
+UNIFORM = 'uniform'
+OFF = 'off'
+MYSQL_ROOT_USER = 'root'
+MYSQL_ROOT_PASSWORD_PREFIX = 'Perfkit8'
+
+PREPARE_SCRIPT_PATH = '/usr/share/doc/sysbench/tests/db/parallel_prepare.lua'
+OLTP_SCRIPT_PATH = '/usr/share/doc/sysbench/tests/db/oltp.lua'
+
+SYSBENCH_RESULT_NAME_DATA_LOAD = 'sysbench data load time'
+SYSBENCH_RESULT_NAME_TPS = 'sysbench tps'
+SYSBENCH_RESULT_NAME_LATENCY = 'sysbench latency'
+NA_UNIT = 'NA'
+SECONDS_UNIT = 'seconds'
+MS_UNIT = 'milliseconds'
+
+# These are the constants that should be specified in GCP's cloud SQL command.
+DEFAULT_BACKUP_START_TIME = '07:00'
+GCP_MY_SQL_VERSION = 'MYSQL_5_6'
+GCP_PRICING_PLAN = 'PACKAGE'
+
+PERCENTILES_LIST = [1, 5, 50, 90, 99, 99.9]
+RESPONSE_TIME_TOKENS = ['min', 'avg', 'max', 'percentile']
+
+
+def GetInfo():
+  return BENCHMARK_INFO
+
+
+class DBStatusQueryError(Exception):
+  pass
+
+
+def _PercentileCalculator(numbers):
+  """Computes percentiles, stddev and mean on a set of numbers
+
+  Args:
+    numbers: The set of numbers to compute percentiles for.
+
+  Returns:
+    A dictionary of percentiles.
+  """
+  numbers_sorted = sorted(numbers)
+  count = len(numbers_sorted)
+  total = sum(numbers_sorted)
+  result = {}
+  for percentile in PERCENTILES_LIST:
+    percentile_string = 'p%s' % str(percentile)
+    result[percentile_string] = numbers_sorted[
+        int(count * float(percentile) / 100)]
+
+  if count > 0:
+    average = total / float(count)
+    result['average'] = average
+    if count > 1:
+      total_of_squares = sum([(i - average) ** 2 for i in numbers])
+      result['stddev'] = (total_of_squares / (count - 1)) ** 0.5
+    else:
+      result['stddev'] = 0
+
+  return result
+
+
+def _GenerateRandomPassword():
+  """ Generates a random password to be used by the DB instance.
+  Args:
+    None
+  Returns:
+    A string that can be used as password to a DB instance.
+  """
+  return '%s%s' % (MYSQL_ROOT_PASSWORD_PREFIX, str(uuid.uuid4())[-8:])
+
+
+def _ParseSysbenchOutput(sysbench_output, results, metadata):
+  """Parses sysbench output and extract relevant TPS and latency numbers, and
```

Comment:
Docstrings should contain a summary line followed by details - see [here](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Comments#Comments).